var/spack/repos/builtin/packages/jsonnet/package.py

```diff
@@ -0,0 +1,26 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack import *
+
+
+class Jsonnet(MakefilePackage):
+    """A data templating language for app and tool developers based on JSON"""
+
+    homepage = "https://jsonnet.org/"
+    git = "https://github.com/google/jsonnet.git"
+
+    maintainers = ["jcpunk"]
+
+    version("develop", branch="master")
+    version("0.18.0", branch="v0.18.0")
+    version("0.17.0", branch="v0.17.0")
+
+    depends_on('gcc@5.5:', type=('build', ), when='@0.18.0:')
```

Comment:
Can you explain this dependencies? Asking since compilers at the moment are node attributes, not real dependencies.

Reply:
Jsonnet 0.18.0 can't be built with gcc older that 5.5.  I was attempting to express that...  Is there a better way to make that explicit?

Reply:
I think for the time being this needs to be expressed with a conflict:
```python
conflicts('%gcc@:5.4.99', when='@0.18.0')
```
For your information, turning compilers into proper dependencies is on the roadmap - but it will take a while to get there. If you want to check a place where `conflicts` are used extensively to express compiler requirements:
```
$ spack edit -b cuda
```

Reply:
@jcpunk My fault, I put a typo above:
```python
conflicts('%gcc@:5.4.99', when='@0.18.0:')
```
There's a `:` at the end to denote an open range

Reply:
update pushed